pkg/network/http/monitor_windows.go

```diff
@@ -0,0 +1,119 @@
+// +build windows,npm
+
+package http
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/network/config"
+	"github.com/DataDog/datadog-agent/pkg/network/driver"
+)
+
+const (
+	defaultMaxTrackedConnections = 65536
+)
+
+// Monitor is responsible for aggregating and emitting metrics based on
+// batches of HTTP transactions received from the driver interface
+type Monitor struct {
+	di         *httpDriverInterface
+	telemetry  *telemetry
+	statkeeper *httpStatKeeper
+
+	mux         sync.Mutex
+	eventLoopWG sync.WaitGroup
+}
+
+// NewMonitor returns a new Monitor instance
+func NewMonitor(c *config.Config) (*Monitor, error) {
+	di, err := newDriverInterface()
+	if err != nil {
+		return nil, err
+	}
+
+	if uint64(c.MaxTrackedConnections) != defaultMaxTrackedConnections {
+		di.setMaxFlows(uint64(c.MaxTrackedConnections))
+	}
+
+	telemetry := newTelemetry()
+
+	return &Monitor{
+		di:         di,
+		telemetry:  telemetry,
+		statkeeper: newHTTPStatkeeper(c.MaxHTTPStatsBuffered, telemetry),
+	}, nil
+}
+
+// Start consuming HTTP events
+func (m *Monitor) Start() {
+	if m == nil {
```

Comment:
Why are we checking the pointer value in each function? It we are using a `nil` `Monitor` somewhere, that feels like a bug and should be fixed.

Reply:
The httpMonitor is `nil` if http monitoring isn't enabled: https://github.com/DataDog/datadog-agent/blob/6db0da8cfbd833ff0a89c876d40a8722d9fa41a2/pkg/network/tracer/tracer_windows.go#L79-L86
FWIW, we do this on the linux side as well.

Reply:
I don't love that pattern, because it relies upon someone remembering to add that to every function. I'd rather take the interface+no-op implementation approach. If you add a function to the interface, you cannot forget to add it to the no-op version because compilation will fail.

Reply:
I like that approach - I'll switch to that on both linux & windows